### PR TITLE
Grant id in mountpoint

### PR DESCRIPTION
**Pull request by micbar**
# Description

We distinguish between an mountpoint and grants for shares.

### CODE REVIEW

File: internal/grpc/services/sharesstorageprovider/sharesstorageprovider.go [hunk 1]
```diff
@@ -687,13 +689,14 @@ func (s *service) Stat(ctx context.Context, req *provider.StatRequest) (*provide
 					OpaqueId:  utils.ShareStorageProviderID,
 				},
 				Type:          provider.ResourceType_RESOURCE_TYPE_CONTAINER,
-				Mtime:         &typesv1beta1.Timestamp{},
+				Mtime:         resourceResp.Info.Mtime,
```
Comment (kobergj) [inline internal/grpc/services/sharesstorageprovider/sharesstorageprovider.go:692]: Just to be sure I got it right: We want the `Mtime` of the resource behind the earliest share, NOT the earliest `Mtime` of all resources behind all shares, right?

File: internal/grpc/services/sharesstorageprovider/sharesstorageprovider.go [hunk 2]
```diff
@@ -939,3 +942,45 @@ func (s *service) rejectReceivedShare(ctx context.Context, receivedShare *collab
...
+		if entry, ok := lsRes.Opaque.Map["etags"]; ok {
+			// If we can't get the etags thats fine, just continue.
+			_ = json.Unmarshal(entry.Value, &shareEtags)
+		}
+	}
+	return receivedShares, shareEtags, nil
+}
+
+func findEarliestShare(receivedShares []*collaboration.ReceivedShare) (earliestShare *collaboration.Share, atLeastOneAccepted bool) {
+	atLeastOneAccepted = false
```
Comment (kobergj) [inline internal/grpc/services/sharesstorageprovider/sharesstorageprovider.go:969]: This line is unnecessary, you have already defined `atLeastOneAccepted` through the named return parameter

**Review by kobergj — CHANGES_REQUESTED** [review]

File: internal/grpc/services/sharesstorageprovider/sharesstorageprovider.go (hunk 1 above)
Comment (micbar) [reply internal/grpc/services/sharesstorageprovider/sharesstorageprovider.go:692]: that is tricky. Need to think about it again. I want the mtime of the earliest share.

**Review by kobergj — APPROVED** [review]

File: internal/grpc/services/sharesstorageprovider/sharesstorageprovider.go (hunk 1 above)
Comment (micbar) [reply internal/grpc/services/sharesstorageprovider/sharesstorageprovider.go]: i changed the implementation. @C0rby FJYI

**Review by kobergj — APPROVED** [review]

File: pkg/utils/utils.go [hunk 3]
```diff
@@ -66,6 +66,12 @@ var (
 	errInvalidSpaceReference = errors.New("invalid storage space reference")
 )
 
+// ShareMetadata contains Metadata for a share
+type ShareMetadata struct {
```
Comment (C0rby) [inline pkg/utils/utils.go:70]: Could you move this to the `share` package and then rename it to just `Metadata`?

**Review by C0rby — APPROVED** [review]
Just one small thing. Apart from that it looks fine.

File: pkg/utils/utils.go (hunk 3 above)
Comment (micbar) [reply pkg/utils/utils.go]: :+1:

**Pull request merged**